phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java

```diff
@@ -550,6 +550,76 @@ public BaseResultIterators(QueryPlan plan, Integer perScanLimit, Integer offset,
         }
         return ranges;
     }
+
+    private List<KeyRange> getRegionRowKeyRanges() throws SQLException {
+        List<HRegionLocation> regionLocations = getRegionBoundaries(scanGrouper);  //Load the region information
+
+        List<KeyRange> regionKeyRanges = Lists.newArrayListWithExpectedSize(regionLocations.size());
+
+        //Map each HRegionLocation to a KeyRange - no Java 8
+        for(HRegionLocation regionLocation : regionLocations){
+            HRegionInfo regionInfo = regionLocation.getRegionInfo();
+
+            //Region is upperInclusive false by definition
+            KeyRange range = KeyRange.getKeyRange(regionInfo.getStartKey(), true, regionInfo.getEndKey(), false);
```

Comment:
Should be

{
	byte[] lowerRange = isFirstRegion ? KeyRange.UNBOUND : regionLocations.get(0).getLowerRange; // Even they are both new byte[0], they are different.
    boolean lowerInclusive = ! isFirstRegion;
	byte[] upperRange = isLastRegion ? KeyRange.UNBOUND : regionLocations.get(regionLocations.size()-1).getUpperRange; // Even they are both new byte[0], they are different.
    boolean upperExclusive = false;
}

Reply:
I agree that we can specifically call out the boundaries I dislike this specific code pasted.  I have added the refactor to specify unbound directly rather than implicitly.